api/utils/keys/policy_test.go

```diff
@@ -14,85 +14,203 @@ See the License for the specific language governing permissions and
-func TestPrivateKeyPolicyError(t *testing.T) {
+func TestIsRequiredPolicyMet(t *testing.T) {
+	privateKeyPolicies := []keys.PrivateKeyPolicy{
+		keys.PrivateKeyPolicyNone,
+		keys.PrivateKeyPolicyHardwareKey,
+		keys.PrivateKeyPolicyHardwareKeyTouch,
+		keys.PrivateKeyPolicyHardwareKeyPIN,
+		keys.PrivateKeyPolicyHardwareKeyTouchAndPIN,
+	}
 	for _, tc := range []struct {
-		desc               string
-		errIn              error
-		expectIsKeyPolicy  bool
-		expectKeyPolicyErr bool
-		expectKeyPolicy    PrivateKeyPolicy
+		requiredPolicy   keys.PrivateKeyPolicy
+		verifiedPolicies []keys.PrivateKeyPolicy
 	}{
 		{
-			desc:               "random error",
-			errIn:              trace.BadParameter("random error"),
-			expectIsKeyPolicy:  false,
-			expectKeyPolicyErr: true,
+			requiredPolicy:   keys.PrivateKeyPolicyNone,
+			verifiedPolicies: privateKeyPolicies,
 		}, {
-			desc:               "unknown_key_policy",
-			errIn:              NewPrivateKeyPolicyError("unknown_key_policy"),
-			expectIsKeyPolicy:  true,
-			expectKeyPolicyErr: true,
+			requiredPolicy:   keys.PrivateKeyPolicyHardwareKey,
+			verifiedPolicies: hardwareKeyPolicies,
 		}, {
-			desc:              string(PrivateKeyPolicyNone),
-			errIn:             NewPrivateKeyPolicyError(PrivateKeyPolicyNone),
-			expectIsKeyPolicy: true,
-			expectKeyPolicy:   PrivateKeyPolicyNone,
+			requiredPolicy:   keys.PrivateKeyPolicyHardwareKeyTouch,
+			verifiedPolicies: hardwareKeyTouchPolicies,
 		}, {
-			desc:              string(PrivateKeyPolicyHardwareKey),
-			errIn:             NewPrivateKeyPolicyError(PrivateKeyPolicyHardwareKey),
-			expectIsKeyPolicy: true,
-			expectKeyPolicy:   PrivateKeyPolicyHardwareKey,
+			requiredPolicy:   keys.PrivateKeyPolicyHardwareKeyPIN,
+			verifiedPolicies: hardwareKeyPINPolicies,
 		}, {
-			desc:              string(PrivateKeyPolicyHardwareKeyTouch),
-			errIn:             NewPrivateKeyPolicyError(PrivateKeyPolicyHardwareKeyTouch),
-			expectIsKeyPolicy: true,
-			expectKeyPolicy:   PrivateKeyPolicyHardwareKeyTouch,
+			requiredPolicy:   keys.PrivateKeyPolicyHardwareKeyTouchAndPIN,
+			verifiedPolicies: hardwareKeyTouchAndPINPolicies,
+		},
+	} {
+		t.Run(string(tc.requiredPolicy), func(t *testing.T) {
+			for _, keyPolicy := range privateKeyPolicies {
+				if keys.IsRequiredPolicyMet(tc.requiredPolicy, keyPolicy) {
+					require.Contains(t, tc.verifiedPolicies, keyPolicy, "Policy %q does not meet %q but IsRequirePolicyMet(%v, %v) returned true", keyPolicy, tc.requiredPolicy, tc.requiredPolicy, keyPolicy)
+				} else {
+					require.NotContains(t, tc.verifiedPolicies, keyPolicy, "Policy %q does meet %q but IsRequirePolicyMet(%v, %v) returned false", keyPolicy, tc.requiredPolicy, tc.requiredPolicy, keyPolicy)
+				}
+			}
+		})
+	}
+}
+
+func TestGetPolicyFromSet(t *testing.T) {
+	testCases := []struct {
+		name            string
+		policySet       []keys.PrivateKeyPolicy
+		expectSetPolicy keys.PrivateKeyPolicy
```

Comment:
nit: wantPolicy, in keeping with Go got/want lingo?

```suggestion
		wantPolicy keys.PrivateKeyPolicy
```